lib/dirac_domain_wall_4d.cpp

```diff
@@ -158,42 +155,38 @@ namespace quda {
   {
     if ( in.Ndim() != 5 || out.Ndim() != 5) errorQuda("Wrong number of dimensions\n");
     double kappa2 = kappa5*kappa5;
-
-    bool reset1 = newTmp(&tmp1, in);
+    auto tmp = getFieldTmp(in);
 
     int odd_bit = (matpcType == QUDA_MATPC_ODD_ODD || matpcType == QUDA_MATPC_ODD_ODD_ASYMMETRIC) ? 1 : 0;
     bool symmetric =(matpcType == QUDA_MATPC_EVEN_EVEN || matpcType == QUDA_MATPC_ODD_ODD) ? true : false;
     QudaParity parity[2] = {static_cast<QudaParity>((1 + odd_bit) % 2), static_cast<QudaParity>((0 + odd_bit) % 2)};
 
     if (symmetric && !dagger) {
       // 1 - k^2 M5^-1 D4 M5^-1 D4
-      Dslash4(*tmp1, in, parity[0]);
-      M5inv(out, *tmp1);
-      Dslash4(*tmp1, out, parity[1]);
-      M5invXpay(out, *tmp1, in, -kappa2);
+      Dslash4(tmp, in, parity[0]);
+      M5inv(out, tmp);
+      Dslash4(tmp, out, parity[1]);
+      M5invXpay(out, tmp, in, -kappa2);
     } else if (symmetric && dagger) {
       // 1 - k^2 D4 M5^-1 D4 M5^-1
-      M5inv(*tmp1, in);
-      Dslash4(out, *tmp1, parity[0]);
-      M5inv(*tmp1, out);
-      Dslash4Xpay(out, *tmp1, parity[1], in, -kappa2);
+      M5inv(tmp, in);
+      Dslash4(out, tmp, parity[0]);
+      M5inv(tmp, out);
+      Dslash4Xpay(out, tmp, parity[1], in, -kappa2);
     } else {
       // 1 - k D5 - k^2 D4 M5^-1 D4_oe
-      Dslash4(*tmp1, in, parity[0]);
-      M5inv(out, *tmp1);
-      Dslash4Xpay(*tmp1, out, parity[1], in, -kappa2);
-      Dslash5Xpay(out, in, *tmp1, -kappa5);
+      Dslash4(tmp, in, parity[0]);
+      M5inv(out, tmp);
+      Dslash4Xpay(tmp, out, parity[1], in, -kappa2);
+      Dslash5Xpay(out, in, tmp, -kappa5);
     }
-
-    deleteTmp(&tmp1, reset1);
   }
 
   void DiracDomainWall4DPC::MdagM(ColorSpinorField &out, const ColorSpinorField &in) const
   {
-    bool reset = newTmp(&tmp2, in);
-    M(*tmp2, in);
-    Mdag(out, *tmp2);
-    deleteTmp(&tmp2, reset);
+    auto tmp = getFieldTmp(in);
+    M(tmp, in);
+    Mdag(out, tmp);
```

Comment:
This is great in the sense that we do not need to distinguish between `tmp1` and `tmp2` anymore. :)